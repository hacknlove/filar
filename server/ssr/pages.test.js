const { loadPage, getPage } = require('./pages');
const { readFile } = require('fs-extra');

jest.mock('fs-extra', () => ({
    readFile: jest.fn(() => Promise.resolve()),
}));

describe('getPage', () => {
    test('should throw an error if the page is not found', async () => {
        await expect(() => getPage('/path/to/nonexistent/page.html')).rejects.toThrow('Page not found');
    });

    test('should return a cloned node if the page exists', async () => {
        readFile.mockResolvedValueOnce('<html><body>Hello World!</body></html>');
        const filePath = '/path/to/existing/page.html';
        await loadPage(filePath);
        const clonedNode = await getPage(filePath);
        expect(clonedNode.toString()).toMatchSnapshot();
    });
});