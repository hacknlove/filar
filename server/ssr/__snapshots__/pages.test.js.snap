// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPage should return a cloned node if the page exists 1`] = `"<html><body>Hello World!</body></html>"`;
